File: mobile-expo/src/config/env.ts
import Constants from 'expo-constants';

/**
 * API base URL, read from app.json's `expo.extra.apiBaseUrl`, or overridden
 * at build time with an EAS build profile / environment variable.
 *
 * - Physical device with Expo Go: use your computer's LAN IP, not localhost
 *   (e.g. http://192.168.1.23:8000) — the phone can't resolve "localhost"
 *   as your dev machine.
 * - Android emulator: http://10.0.2.2:8000
 * - iOS simulator: http://localhost:8000 works fine.
 */
export const API_BASE_URL: string =
  (Constants.expoConfig?.extra?.apiBaseUrl as string) || 'http://localhost:8000';

File: mobile/lib/config/env.dart
/// API base URL, overridable at build/run time:
///   flutter run --dart-define=API_BASE_URL=https://api.yourdomain.com
///   flutter build apk --dart-define=API_BASE_URL=https://api.yourdomain.com
///
/// Defaults to a local backend for development.
/// - Android emulator reaches the host machine via 10.0.2.2, not localhost.
/// - iOS simulator can use localhost directly.
class Env {
  static const String apiBaseUrl = String.fromEnvironment(
    'API_BASE_URL',
    defaultValue: 'http://10.0.2.2:8000',
  );
}
